astropy/wcs/tests/test_prjprm.py

```diff
@@ -0,0 +1,237 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from copy import copy, deepcopy
+import pytest
+
+import numpy as np
+from astropy import wcs
+
+from . helper import SimModelTAB
+
+
+def test_prjprm_init():
+    # test PyPrjprm_cnew
+    assert wcs.WCS().wcs.cel.prj
+
+    # test PyPrjprm_new
+    assert wcs.Prjprm()
+
+    with pytest.raises(wcs.InvalidPrjParametersError):
+        prj = wcs.Prjprm()
+        prj.set()
+
+    # test deletion does not crash
+    prj = wcs.Prjprm()
+    del prj
+
+
+def test_prjprm_copy():
+    # shallow copy
+    prj = wcs.Prjprm()
+    prj2 = copy(prj)
+    prj3 = copy(prj2)
+    prj.pv = [0, 6, 8, 18, 3]
+    assert (np.allclose(prj.pv, prj.pv, atol=1e-12, rtol=0) and
+            np.allclose(prj.pv, prj.pv, atol=1e-12, rtol=0))
+    del prj, prj2, prj3
+
+    # deep copy
+    prj = wcs.Prjprm()
+    prj2 = deepcopy(prj)
+    prj.pv = [0, 6, 8, 18, 3]
+    assert not np.allclose(prj.pv, prj2.pv, atol=1e-12, rtol=0)
+    del prj, prj2
+
+
+def test_prjprm_flag():
+    prj = wcs.Prjprm()
+    assert prj._flag == 0
+
+
+def test_prjprm_code():
+    prj = wcs.Prjprm()
+
+    assert prj.code == '   '
+    assert prj._flag == 0
+
+    prj.code = 'TAN'
+    prj.set()
+    assert prj.code == 'TAN'
+    assert prj._flag
+
+    prj.code = 'TAN'
+    assert prj._flag
+
+    prj.code = None
+    assert prj.code == '   '
+    assert prj._flag == 0
+
+
+def test_prjprm_phi0():
+    prj = wcs.Prjprm()
+
+    assert prj.phi0 == None
+    assert prj._flag == 0
+
+    prj.code = 'TAN'
+    prj.phi0 = 2.0
+    prj.set()
+    assert prj.phi0 == 0
+
+    prj.phi0 = 0.0
+    assert prj._flag
+
+    prj.phi0 = 2.0
+    assert prj._flag == 0
+
+    prj.phi0 = None
+    assert prj.phi0 == None
+    assert prj._flag == 0
+
+
+def test_prjprm_theta0():
+    prj = wcs.Prjprm()
+
+    assert prj.theta0 == None
+    assert prj._flag == 0
+
+    prj.code = 'TAN'
+    prj.phi0 = 2.0
+    prj.theta0 = 4.0
+    prj.set()
+    assert prj.theta0 == 4.0
+
+    prj.theta0 = 4.0
+    assert prj._flag
+
+    prj.theta0 = 8.0
+    assert prj._flag == 0
```

Comment:
According to the [docs](https://www.atnf.csiro.au/people/mcalabre/WCS/wcslib/structprjprm.html#ad304d66b3f3aa64fe9c7251d3c420d02),

> This flag must be set to zero whenever any of the following prjprm struct members are set or changed:
> 
> prjprm::code,
> prjprm::r0,
> prjprm::pv[],
> prjprm::phi0,
> prjprm::theta0.

So, because `prj.theta0` has changed it's value, `flag` should be set (by the setter) to 0. This test checks that indeed `flag` is 0 on changing `theta0` value.